From d0a9d78792970d2a1c25d35d3a6412d5c5fe9135 Mon Sep 17 00:00:00 2001
From: "Thakur, Santosh Krx" <santosh.krx.thakur@intel.com>
Date: Thu, 4 Feb 2016 19:39:44 +0530
Subject: [PATCH 24/24] SoC: Intel: BYT: klockwork error in Audio Driver

This patch checks for NULL before dereferencing "conf".

Change-Id: If41f3608d5ea0de83da784fe2ad6d65e701e6e1b
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-18600
Signed-off-by: Thakur, Santosh Krx <santosh.krx.thakur@intel.com>
Reviewed-on: https://android.intel.com:443/466555
---
 sound/soc/intel/board/byt_cr_dpcm_rt5651.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/sound/soc/intel/board/byt_cr_dpcm_rt5651.c b/sound/soc/intel/board/byt_cr_dpcm_rt5651.c
index 04ecb47..0461c8d 100644
--- a/sound/soc/intel/board/byt_cr_dpcm_rt5651.c
+++ b/sound/soc/intel/board/byt_cr_dpcm_rt5651.c
@@ -895,6 +895,11 @@ static int snd_byt_mc_probe(struct platform_device *pdev)
 
 	/* Get board-specific HW-settings */
 	conf = get_board_config(get_mc_link());
+	if (!conf) {
+		pr_err("%s: Failed to get board-specific HW-settings!\n",
+			__func__);
+		return -EFAULT;
+	}
 	switch (conf->idx) {
 	case RT5651_ANCHOR8:
 	case RT5651_DEFAULT:
-- 
1.9.1

